lib/Extractor/Solver.cpp

```diff
@@ -76,6 +76,157 @@ class BaseSolver : public Solver {
+    bool IsSat;
+    std::error_code EC = SMTSolver->isSatisfiable(BuildQuery(IC, BPCs, PCs, Mapping, 0, /*Precondition=*/0),
+                                                  IsSat, 0, 0, Timeout);
+    if (EC)
+      llvm::report_fatal_error("stopping due to error");
+    return !IsSat;
+  }
+
+  std::error_code negative(const BlockPCs &BPCs,
+                           const std::vector<InstMapping> &PCs,
+                           Inst *LHS, bool &Negative,
+                           InstContext &IC) override {
+    unsigned W = LHS->Width;
+    Negative = false;
+    if (testZeroMSB(BPCs, PCs, LHS, IC))
+      Negative = false;
+    else if (testOneMSB(BPCs, PCs, LHS, IC))
+      Negative = true;
+    return std::error_code();
+  }
+
+  std::error_code nonNegative(const BlockPCs &BPCs,
+                              const std::vector<InstMapping> &PCs,
+                              Inst *LHS, bool &NonNegative,
+                              InstContext &IC) override {
+    unsigned W = LHS->Width;
+    NonNegative = false;
+    if (testZeroMSB(BPCs, PCs, LHS, IC)) {
+      NonNegative = true;
+    } else if (testOneMSB(BPCs, PCs, LHS, IC)) {
+      NonNegative = false;
+    }
+    return std::error_code();
+  }
+
+  std::error_code knownBits(const BlockPCs &BPCs,
+                          const std::vector<InstMapping> &PCs,
+                          Inst *LHS, KnownBits &Known,
+                          InstContext &IC) override {
+    unsigned W = LHS->Width;
+    Known.One = APInt::getNullValue(W);
+    Known.Zero = APInt::getNullValue(W);
+    for (unsigned I=0; I<W; I++) {
+      APInt ZeroGuess = Known.Zero | APInt::getOneBitSet(W, I);
+      if (testKnown(BPCs, PCs, ZeroGuess, Known.One, LHS, IC)) {
+        Known.Zero = ZeroGuess;
+        continue;
+      }
+      APInt OneGuess = Known.One | APInt::getOneBitSet(W, I);
+      if (testKnown(BPCs, PCs, Known.Zero, OneGuess, LHS, IC))
+        Known.One = OneGuess;
+    }
+    return std::error_code();
+  }
+
+  std::error_code powerTwo(const BlockPCs &BPCs,
+                           const std::vector<InstMapping> &PCs,
+                           Inst *LHS, APInt &PowTwo,
+                           InstContext &IC) override {
+    unsigned W = LHS->Width;
+    Inst *PowerMask = IC.getInst(Inst::And, W, {LHS, IC.getInst(Inst::Sub, W, {LHS, IC.getConst(APInt(W, 1, false))})});
+    Inst *Zero = IC.getConst(APInt(W, 0, false));
+    Inst *True = IC.getConst(APInt(1, 1, false));
+    Inst *PowerTwoInst = IC.getInst(Inst::And, 1, {IC.getInst(Inst::Ne, 1, {LHS, Zero}),
+                                                   IC.getInst(Inst::Eq, 1, {PowerMask, Zero})});
+    InstMapping Mapping(PowerTwoInst, True);
+    bool IsSat;
+    std::error_code EC = SMTSolver->isSatisfiable(BuildQuery(IC, BPCs, PCs, Mapping, 0, /*Precondition=*/0),
+                                                  IsSat, 0, 0, Timeout);
+    if (EC)
+      llvm::report_fatal_error("stopping due to error");
```

Comment:
in this case and others, say something useful when an error occurs, ```stopping due to error``` is not useful to someone running this code